lib/wrap-request.js

```diff
@@ -0,0 +1,59 @@
+/* eslint padded-blocks: 0 */
+module.exports = wrapRequest
+
+const noop = () => Promise.resolve()
+
+function wrapRequest (state, emitter, request, options, retries = 0) {
+  const retryRequest = function (retryAfter) {
+    return new Promise(resolve => setTimeout(resolve, retryAfter * 1000))
+      .then(() => wrapRequest(state, emitter, request, options, retries + 1))
+  }
+
+  const jobOptions = retries > 0 ? { weight: 0, priority: 0 } : {}
+  return state.globalLimiter.schedule(jobOptions, async function () {
+
+    // Guarantee at least 1000ms between writes
+    if (options.method !== 'GET' && options.method !== 'HEAD') {
+      await state.writeLimiter.schedule(noop)
+    }
+
+    // Guarantee at least 5000ms between requests that trigger notifications
+    if (state.triggersNotification[options.url]) {
+      await state.triggersNotificationLimiter.schedule(noop)
+    }
+
+    try {
+      // Execute request
+      return await request(options)
+    } catch (error) {
+      if (error.status === 403 && error.message.toLowerCase().includes('abuse')) {
```

Comment:
I can't think of a word that includes "abuse", but maybe to avoid false positive, we could check with a regex?

```suggestion
      if (error.status === 403 && /\babuse\b/i.test(error.message)) {
```

Reply:
Sure! I'm changing it.